bpf/gotracer/go_sql.c

```diff
@@ -25,11 +25,99 @@
 #include <common/ringbuf.h>
 
 #include <gotracer/go_common.h>
+#include <gotracer/go_str.h>
+
+// Extracts MySQL server hostname by following the pointer chain:
+// driverConn -> ci (interface) -> mysqlConn -> cfg (*Config) -> Addr (string)
+// Returns the Addr string which has format "host:port"
+static __always_inline int
+read_mysql_hostname_from_driverconn(u64 driver_conn_ptr, char *hostname, int max_len) {
+    if (driver_conn_ptr == 0) {
+        bpf_dbg_printk("can't read mysql hostname: driver_conn_ptr is null");
+        return 0;
+    }
+
+    off_table_t *ot = get_offsets_table();
```

Comment:
what's the reason we check for nullness of the offset table?

Reply:
it's too verbose and we have log errors when we can't register an offsets table. we should remove those lines.